docs/contributing/adding-images-to-the-wesite.md

```diff
@@ -0,0 +1,27 @@
+# Adding images to the design system website
+When adding or amending guidance on the design system website and you want to add an image, it's important to be considerate of the impact images have on the website. They can have implications for web performance and accessibility if used improperly.
+
+Before reading the following, also consult our [image guidance](https://design-system.service.gov.uk/styles/images/) for details on when it's appropriate to use images and how best to use them.
+
+## Does your content really need images?
+Ideally, we would avoid adding new images to our content.
+
+In patterns and components, we should be using code examples when we want to convey what our pattern or component will look like. Avoid using screenshots of patterns or components wherever possible.
+
+## Can your image be an SVG?
+SVG's are better for web performance as they scale with different screen sizes and load faster than most image formats of the same size. If your image is a "graphic" aka: it isn't a photo or a screenshot, it should be converted to an SVG and added to the page directly.
+
+## Ensure your image is an appropriate file size
+Avoid adding very large file size images to the website.
+
+If your image is larger than the context it is presented in eg: if you have an image with an intrinsic size of 1200x1200 pixels but the max width it will ever be displayed at on screens is 500 pixels, it should be resized so that it's intrinsically 500 pixels wide. You can discover what size an image is rendered on a given screen size by opening your browser's dev tools, locating the `img` tag which is rendering the image and hovering over the `src` of that tag. You can then move the screen size down to assess at what point the image is largest. Don't assume that the image is largest on desktop only as reflowing layouts can mean that a given image is filling the screen and therefore at it's largest on mobile.
```

Comment:
I'm not sure if the 500px width recommendation applies to SVGs (see [1400px `task-list-whole.svg`](https://github.com/alphagov/govuk-design-system/pull/2577)) as this may not have any affect on browser performance when the image is rendered into pixels

Reply:
Also need to bear in mind that "500px" on a HiDPI (High Dots Per Inch) display—or when using an increased scaling mode for readability—might take up 1000px `@2x` or 1500px `@3x` or any number of pixels

Nice crisp images start to look blurry

Where SVGs can't be used, this has been solved with `srcset`

Comment brought over from:

* https://github.com/alphagov/govuk-design-system/pull/2577#issuecomment-1412018196

## Scaling
For scenarios where an image is preferable due to file size (versus an SVG vector graphic) we should remember that an image will look blurry when zooming in or when displayed on "high DPI" (dots per inch) screens

i.e. Pixels aren't scalable like vectors

Might be a good test for the new 500px guideline size, would we need to double (2x) or triple (3x) the pixel size as device DPI changes? Can image compression also be increased since larger pixel images take up the same viewport size?

```html
<img
  src="example.png"
  srcset="example.png, example@2x.png 2x, example@3x.png 3x"
  alt="A screenshot showing an example."
>
```

**Note:** Blurry images might be preferred to longer load times with larger KB images

Reading material: https://web.dev/high-dpi/

Reply:
I wrote this on the assumption that at this point, the person had already made the calculation on if they should use an SVG over a PNG. I wonder if I can filter some of that into the SVG section though? Also from your feedback it sounds like you interpreted the 500px thing as an explicit recommendation when it was just an example to illustrate not going above maximum width in a given context. I'm gonna see if I can make this clearer.

You do raise a good point though on high quality displays like retina macs. We actually found during the community page work that a properly optimised image that's double the size of the max width is a lot better than a poorly optimised image of the same size.